src/map/entities/petentity.cpp

```diff
@@ -206,12 +237,58 @@ void CPetEntity::Spawn()
         mobutils::GetAvailableSpells(this);
     }
 
+    if (m_PetType == PET_TYPE::JUG_PET)
+    {
+        m_jugSpawnTime = server_clock::now();
```

Comment:
This is ok to do on spawn because pets override spawn time / other vars after spawn with LoadPet() in petutils. I don't really like these intrinsic state changes but the pet zoning system has a few.